languageserver/integration/codelenses.go

```diff
@@ -83,23 +98,64 @@ func (i *FlowIntegration) showDeployContractAction(
 	}
 
 	name := contract.Identifier.Identifier
-
 	position := contract.StartPosition()
+	codelensRange := conversion.ASTToProtocolRange(position, position)
+	var codeLenses []*protocol.CodeLens
 
-	return &protocol.CodeLens{
-		Range: conversion.ASTToProtocolRange(
-			position,
-			position,
-		),
-		Command: &protocol.Command{
-			Title: fmt.Sprintf(
-				"deploy to account 0x%s",
-				i.activeAddress.Hex(),
-			),
-			Command:   CommandDeployContract,
-			Arguments: []interface{}{uri, name},
-		},
+	// If emulator is not up, we need to provide actionless codelens
+	if i.emulatorState == EmulatorStarting || i.emulatorState == EmulatorOffline {
+		return codeLenses
 	}
+
+	signersList := entryPointInfo.pragmaSignersStrings[:]
+	// TODO: resolve check for amount of signers equals to provided by pragma
+	if len(signersList) == 0 {
+		activeAccount := [][]string{[]string{"no-signers"}}
+		signersList = append(signersList, activeAccount...)
+	}
+
+	for _, signers := range signersList {
+		signer := signers[0]
+		if signer == "no-signers" {
```

Comment:
Instead of using and comparing with a magic string, maybe e.g. just initialize to `[][]string{[]string{}` and check the length

Reply:
I decided to simply add single account there from the start :)
That was a bit of a silly approach...